easybuild/easyblocks/generic/cmakemake.py

```diff
@@ -209,16 +210,22 @@ def configure_step(self, srcdir=None, builddir=None):
         options.append('-DCMAKE_FIND_USE_PACKAGE_REGISTRY=FALSE')
 
         if not self.cfg.get('allow_system_boost', False):
-            # don't pick up on system Boost if Boost is included as dependency
-            # - specify Boost location via -DBOOST_ROOT
-            # - instruct CMake to not search for Boost headers/libraries in other places
-            # - disable search for Boost CMake package configuration file
             boost_root = get_software_root('Boost')
             if boost_root:
+                # Check for older builds of Boost
+                cmake_files = glob.glob(os.path.join(boost_root, 'lib', 'cmake', 'boost_system*',
+                                                     'libboost_system-variant*-shared.cmake'))
+                if len(cmake_files) > 1 and 'libboost_system-variant-shared.cmake' in cmake_files:
```

Comment:
This check is incorrect, you should only be checking the filename of the entries in `cmake_files` when checking whether `'libboost_system-variant-shared.cmake'` is included?

With the `CMakeMake` easyblock from this PR, the installation of `SOCI-4.0.1-GCC-10.2.0.eb` is broken when `Boost` was not reinstalled with the updated easyblock from #2456.

Contents of `$EBROOTBOOST/lib/cmake/boost_system-1.74.0`:
```
boost_system-config.cmake
boost_system-config-version.cmake
libboost_system-variant-mt-x64-shared.cmake
libboost_system-variant-mt-x64-static.cmake
libboost_system-variant-shared.cmake
libboost_system-variant-static.cmake
```

Reply:
fixed in 22b339c05

Reply:
Ah yes, didn't think of that for once...